### ISSUE TITLE
Jam Fails with No Connection to Backend 502 error

### ISSUE THREAD

**Issue by Bonzbonanza** [joinmarket-webui/jam#837]
**Expected behavior**

Jam to complete a sweep.

**Actual behavior**

Previously was able to complete a sweep over a day. Trying again and Jam is disconnecting from the backend after only one completed transaction, twice. The first time that this happened today I was able to restart Jam and schedule another sweep, but that has also failed.

**Steps to reproduce the problem**
  1. Run Sweep
  2. Wait
  3.

**Specifications**
  - Version: Synology VM
  - Platform: Umbrel
  - Browser: N/A

**Additional context**

Logs attached
(removed)

**Comment by theborakompanioni** [comment]
Hey @Bonzbonanza. Sorry for the delay and your unpleasant experience.
There is too little info in the logs to actually determine the error cause.
It seems Bitcoin Core is running and is reachable for Jam, but the wallet daemon keeps shutting down.
Would it be possible to inspect the log files in `/var/log/jam`? Please do not include any private or personal information, possibly only the latest error messages :pray:

**Comment by Bonzbonanza** [comment]
Hi @Thebora,

No worries, thanks for the reply. Admittedly, I am very poor at using command line interfaces and had to search for basic Linux commands, but with the Terminal app built into UmbrelOS I don't seem to be able to even find the log. For instance, /var/log returns "Is a directory" but /var/log/jam returns "No such file or directory". Running ls only gives me one file for an SSH key that I created trying to connect during this session. Can you give any insight into how to get the log, or is Umbrel locked down?

**Comment by theborakompanioni** [comment]
> I don't seem to be able to even find the log. For instance, /var/log returns "Is a directory" but /var/log/jam returns "No such file or directory". 

You'd need switch _into_ the Jam container, e.g. with `docker exec -it jam_web_1 bash` (maybe the container is named differently, find out with `docker ps | grep jam` if the previous command did not work).
Then view the log file in `/var/log/jam` with `tail -n 200 -f /var/log/jam/jmwalletd_stdout.log` (and/or `jmwalletd_stderr.log`).
https://jamdocs.org/FAQ/#how-do-i-view-the-log-file

Let me know if that works out for you. Please do not hesitate to write here if it did nor work or does not make sense to you. :pray:

**Comment by Bonzbonanza** [comment]
While the instructions are indeed over my head when though I could probably brute force them if I knew where to enter the commands, the first obstacle is that I'm not using a docker container. My Umbrel is running as it's own VM on a NAS. I'm remote from the NAS which makes it even harder as I access via a browser using TailScale. If you can suggest another method I'm willing to try.

This makes me think that a log output function in the app would be a good idea as it would then be a download in the browser.

**Comment by theborakompanioni** [comment]
> While the instructions are indeed over my head when though I could probably brute force them if I knew where to enter the commands, the first obstacle is that I'm not using a docker container. My Umbrel is running as it's own VM on a NAS. I'm remote from the NAS which makes it even harder as I access via a browser using TailScale. If you can suggest another method I'm willing to try. This makes me think that a log output function in the app would be a good idea as it would then be a download in the browser.

Umbrel uses docker internally to start applications. You'd need to `ssh` into it (as I assume you did earlier when you said that there is no such directory).

There is a "log output function" in the app (for some integrations - those that use the `standalone` image), but it is of no use if the application does not start, damn.. :sweat_smile:

**Comment by Bonzbonanza** [comment]
Thebora,
When trying to execute the commands mentioned ***@***.***:~$ exec -it jam_web_1 bash
-bash: exec: -i: invalid option
exec: usage: exec [-cl] [-a name] [command [argument ...]] [redirection ...]
***@***.***:~$ docker ps | grep jampermission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.44/containers/json": dial unix /var/run/docker.sock: connect: permission denied

I restarted Jam from the Umbrel Home screen and got access to the logs, though they seem to be pretty short. I assume from the restart, which makes me wonder what happens to the older logs.

Logs

User data location: /root/.joinmarket/
2024-09-09 19:00:47,327 [INFO]  Starting jmwalletd on port: 28183
2024-09-09 19:00:47,661 [INFO]  Joinmarket daemon listening on port 27183
<Request at 0x7f0a182a0040 method=POST uri=/api/v1/wallet/JamWallet.jmdat/unlock clientproto=HTTP/1.1>
b'POST'
b'/api/v1/wallet/JamWallet.jmdat/unlock'
{}
b'/api/v1/wallet/JamWallet.jmdat/unlock'
<_io.BytesIO object at 0x7f0a18253ef0>
[(b'Host', [b'100.64.xxx.yyy:5002']), (b'X-Real-Ip', [b'10.21.0.14']), (b'X-Forwarded-For', [b'::ffff:10.21.0.1, 10.21.0.14']), (b'X-Forwarded-Proto', [b'http']), (b'Content-Length', [b'43']), (b'Priority', [b'u=4']), (b'Sec-Gpc', [b'1']), (b'Cookie', [b'_gorilla_csrf=MTcyNTg4OTczMnxJblprT1VOT1VFUTNZVXd5YjNoVVNXWk9PVGhWTUVNclptdG1NR3RXTW13eWFUFl6UTlJZ289fB02r8tM8T_vmq04K5jkr6fmt7VkB7U1h0hFgNq_5UDt; UMBREL_PROXY_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6Icm94eVRva2VuIjp0cnVlLCJpYXQiOjE3MjU5MDc0MjEsImV4cCI6MTcyNjUxMjIyMX0.ITWVJztxazuyYNm5RqJUozlDR751UBGsU8A']), (b'DNT', [b'1']), (b'Origin', [b'http://100.64.xxx.yyy:5002']), (b'Content-Type', [b'text/plain;charset=UTF-8']), (b'Referer', [b'http://100.64.xxx.yyy:5002/']), (b'Accept-Encoding', [b'gzip, deflate']), (b'Accept-Language', [b'en-US,en;q=0.5']), (b'Accept', [b'*/*']), (b'User-Agent', [b'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0']), (b'X-Forwarded-Host', [b'100.64.xxx.yyy:5002'])]
2024-09-09 19:01:24,319 [WARNING]  Connection had broken pipe, attempting reconnect.
2024-09-09 19:01:28,655 [INFO]  Starting transaction monitor in walletservice

**Comment by theborakompanioni** [comment]
So, it seems that it is sorted out? Or does it keep happening?
Is there anything else you did before it the hiccups started?
On integrations like Umbrel Jam tries to only start if Bitcoin Core accepts RPC requests and is completely synced. Maybe there is a chance this is not working properly on every Core version.. :thinking:

**Comment by Bonzbonanza** [comment]
No, same problems. Core just updated to 27.1 last night after using Jam to do a couple quick Conjoins, then my computer that I use to connect to the Umbrel via a browser rebooted overnight. This morning there's the same error. Restarting the app is only a temporary fix that seems to work while then using it, but during the Sweep operation it always fails.

-------- Original Message --------

**Comment by theborakompanioni** [comment]
I am sorry I cannot identify the error based on the description : /
Do single collaborative transactions or direct-spends work normally?

**Comment by Bonzbonanza** [comment]
Yes, single collaborative transactions generally work, though some have taken a couple tries. Spends are no issue.

-------- Original Message --------

**Comment by theborakompanioni** [comment]
Hmm.. still scratching my head about this. I am really shooting in the dark here wondering what might be the culprit.. e.g. Have you updated Bitcoin Core recently?
A new version should be released shortly. Hopefully your situation improves but this would be a more than lucky coincidence.

**Comment by kn0wmad** [comment]
I also got this problem - seems like tor connectivity is the culprit, but repeated restarts don't seem to remedy it.  

Some relevant logs:

```
2025-03-08T16:37:00+00:00  2025-03-08 16:37:00,023 [INFO]  We, NOT-SERVING-ONION, are calling the handshake callback as client.
2025-03-08T16:37:00+00:00  2025-03-08 16:37:00,023 [INFO]  Sending this handshake: {"app-name": "joinmarket", "directory": false, "location-string": "NOT-SERVING-ONION", "proto-ver": 5, "features": {}, "nick": "******", "network": "mainnet"} to peer ******************************.onion:5222
2025-03-08T16:37:00+00:00  2025-03-08 16:37:00,024 [INFO]  Going to reattempt connection to ******************************.onion:5222 in 10.0 seconds.
2025-03-08T16:37:01+00:00  2025-03-08 16:37:01,862 [INFO]  Failed to connect to peer ******************************.onion:5222
2025-03-08T16:37:01+00:00  2025-03-08 16:37:01,863 [INFO]  Going to reattempt connection to ******************************.onion:5222 in 6.0 seconds.
2025-03-08T16:37:10+00:00  2025-03-08 16:37:10,430 [INFO]  Updating status to connected for peer ******************************.onion:5222.
2025-03-08T16:37:10+00:00  2025-03-08 16:37:10,430 [INFO]  We, NOT-SERVING-ONION, are calling the handshake callback as client.
2025-03-08T16:37:10+00:00  2025-03-08 16:37:10,431 [INFO]  Sending this handshake: {"app-name": "joinmarket", "directory": false, "location-string": "NOT-SERVING-ONION", "proto-ver": 5, "features": {}, "nick": "********", "network": "mainnet"} to peer ******************************.onion:5222
```

**Comment by 86codyharris** [comment]
i'm gettting the same as above using a Start9 server running Knots.  

single input spends work
collaboarative spends dont work
Earn mixes work sometimes, but not very often

wondering if reverting to last version would be a worthwhile test?

**Comment by theborakompanioni** [comment]
Hey @86codyharris.

Thanks for the heads-up. Let's see if a new JM version fixes things.

**Comment by 86codyharris** [comment]
> Hey [@86codyharris](https://github.com/86codyharris).
> 
> Thanks for the heads-up. Let's see if a new JM version fixes things.

Any idea when this new JM version is expected?  I would say that for me JAM is currently unusable(restarts don't reliably help)  And I suspect unusable for other Start9 users.

**Comment by theborakompanioni** [comment]
> > Hey [@86codyharris](https://github.com/86codyharris).
> > Thanks for the heads-up. Let's see if a new JM version fixes things.
> 
> Any idea when this new JM version is expected? I would say that for me JAM is currently unusable(restarts don't reliably help) And I suspect unusable for other Start9 users.

ping @kristapsk

**Comment by theborakompanioni** [comment]
Jam v0.4.0 is released which hopefully improves this.

Closing! Please reopen if you think this is relevant.
